File: AccoMind.Frontend/src/types/company.ts
export type CompanyType = {
    id: number;
    name: string;
    stock_symbol: string;
    logo_url: string;
    sector: string;
}

type CompanyMarketInfoType = {
    stock_price: number;
    price_change: number;
    market_cap: number;
    created_at: string;
}

type CompanyMetricsType = {
    pe_ratio: number;
    dividend_yield: number;
    beta: number;
}

export type CompanyByIdType = {
    id: number;
    name: string;
    stock_symbol: string;
    logo_url: string;
    description: string;
    industry: string;
    founded: number;
    location: string;
    website: string;
    market_info: CompanyMarketInfoType;
    metrics: CompanyMetricsType;
}

export type CompanyMarket = {
    stock_price: number;
    price_change: number;
    market_cap: number;
}

File: Backend/schemas/company.py
import datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, PastDatetime

class CompanyCreateSchema(BaseModel):
    name: str
    description: str

class MarketInfoSchema(BaseModel):
    market_cap: Decimal
    price_change: Decimal
    stock_price: float
    created_at: PastDatetime

class MetricsSchema(BaseModel):
    pe_ratio: Decimal
    dividend_yield: Decimal
    beta: Optional[Decimal]


class CompanySchema(BaseModel):
    id: int
    name: str
    description: str
    stock_symbol: str
    logo_url: str
    industry: str
    location: str | None
    founded: int | None
    # market_info: Optional[MarketInfoSchema]

    class Config:
        from_attributes = True

class CompanyByIdSchema(CompanySchema):
    market_info: Optional[MarketInfoSchema]
    metrics: Optional[MetricsSchema]